src/main/java/org/dependencytrack/tasks/vulnerabilitypolicy/AbstractVulnerabilityPolicyFetcher.java

```diff
@@ -0,0 +1,234 @@
+                        return false;
+                    }
+                }
+                zipEntry = zipInputStream.getNextEntry();
+            }
+        }
+        return true;
+
+    }
+
+    @Override
+    public boolean parseAndSaveVulnerabilityPolicy(StringBuilder out) {
+        ObjectMapper objMapper = new ObjectMapper(new YAMLFactory());
+        final File jsonSchemaFile = new File("src/main/java/org/dependencytrack/policy/validation/policySchema.json");
+        final URI uri = jsonSchemaFile.toURI();
+        JsonSchemaFactory factory = JsonSchemaFactory.builder(JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012)).objectMapper(objMapper).build();
+        JsonSchema schema = factory.getSchema(uri);
+        JsonNode jsonNode = null;
+        try {
+            jsonNode = objMapper.readTree(out.toString());
+        } catch (IOException ex) {
+            logger.error("Unable to parse provided yaml into json content", ex);
+            return false;
+        }
+        Set<ValidationMessage> validateMsg = schema.validate(jsonNode);
+        try (QueryManager queryManager = new QueryManager()) {
+            List<VulnerabilityPolicy> vulnerabilityPolicies = queryManager.getAllVulnerabilityPolicies();
+            List<String> policyNames = vulnerabilityPolicies.stream().map(VulnerabilityPolicy::getName).toList();
+            if (validateMsg.isEmpty()) { //yaml file provided is valid and we can proceed to validate the cel policy
+                ObjectMapper objectMapper = new ObjectMapper();
+                ArrayList vulnerabilityCelPolicyConditions = objectMapper.convertValue(jsonNode.get("conditions"), ArrayList.class);
+                try {
+                    validateVulnerabilityCelPolicy(vulnerabilityCelPolicyConditions);
+                } catch (ScriptCreateException ex) {
+                    logger.error("There was an error in parsing the specified cel policy in the yaml document", ex);
+                    return false;
+                }
+                VulnerabilityPolicy vulnerabilityPolicy = createVulnerabilityPolicyObject(objectMapper, jsonNode, vulnerabilityCelPolicyConditions);
+                if (vulnerabilityPolicy != null) {
+                    if (policyNames.contains(vulnerabilityPolicy.getName())) {
+                        return queryManager.updateVulnerablePolicyByName(vulnerabilityPolicy) == 1;
+                    } else {
+                        return queryManager.createVulnerabilityPolicy(vulnerabilityPolicy) == 1;
+                    }
+                }
+            }
+            return false;
+        }
+    }
+
+    private VulnerabilityPolicy createVulnerabilityPolicyObject(ObjectMapper objectMapper, JsonNode jsonNode, ArrayList vulnerabilityCelPolicyConditions) {
+        try {
+            VulnerabilityPolicy vulnerabilityPolicy = new VulnerabilityPolicy();
+            vulnerabilityPolicy.setName(jsonNode.get("name").asText());
```

Comment:
can we use JacksonMapper convertTreeToValue to convert it to POJO object. It might reduce code